src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1305,9 +1308,14 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		if (this._isDisposed) {
 			return;
 		}
-
-		const trusted = await this._trust();
-		if (!trusted) {
+		const activeWorkspaceRootUri = this._historyService.getLastActiveWorkspaceRoot(Schemas.file);
+		if (activeWorkspaceRootUri) {
+			const trusted = await this._trust();
+			if (!trusted) {
+				this._onProcessExit({ message: nls.localize('workspaceNotTrustedCreateTerminal', "Cannot launch a terminal process in an untrusted workspace") });
+			}
+		} else if (this._cwd !== this._userHome) {
+			// something strange is going on if cwd is not userHome in an empty workspace
 			this._onProcessExit({ message: nls.localize('workspaceNotTrustedCreateTerminal', "Cannot launch a terminal process in an untrusted workspace") });
 		}
```

Comment:
Was expecting this part to look more like:

```
} else {
  this._shellLaunchConfig.cwd = this._userHome;
}
```

Also are we sure `_userHome` is set here as it's a `string | undefined`?